docs/spec/Interfaces.adoc

```diff
@@ -0,0 +1,91 @@
+== Interfaces
+
+Interfaces in F Prime define a set of
+<<Specifiers_Port-Instance-Specifiers,port
+instances>>. Interfaces can be explicitely named
+through a <<Definitions_Port-Interface-Definitions,
+port interface definition>>.
+
+=== Named Interfaces
+
+A named interface is an interface associated
+with an <<Definitions_Port-Interface-Definitions,
+interface definition>>. Named interfaces can be used
+in <<Template-Parameter-Lists,template parameter lists>>.
+
+=== Anonymous Interfaces
+
+There are definitions in FPP that will implicitly define an interface
+so that instances of these definitions may be bound to an interface.
+These implicitly defined interfaces are known as anonymous interfaces
+as they do not have explicit names associated with them.
+
+==== Anonymous Component Interfaces
+
+A <<Definitions_Component-Definitions,component definition>>
+will create a corresponding anonymous interface. This interface includes
+the set of all <<Specifiers_Port-Instance-Specifiers,port instances>>
+inside the component definition (either explicitely, or via an
+<<Specifiers_Interface-Import-Specifiers,interface import>>).
+
+==== Anonymous Topology Interfaces
+
+A <<Definitions_Topology-Definitions,topology definition>>
+will create a corresponding anonymous interface. This interface includes
+the set of all <<Specifiers_Topology-Port-Specifiers,topology port specifiers>>
+inside the topology definition.
+
+=== Instance Values
```

Comment:
Can we avoid using the term "value" here? I don't think there are any values that are computed or used at runtime. Instead, can we use the term "port interface instance" for an instance of a port interface? A port interface instance can be a component instance, a topology, or an interface parameter.